packages/istanbul-lib-instrument/src/visitor.js

```diff
@@ -200,9 +200,99 @@ class VisitState {
             ),
             T.parenthesizedExpression(
                 T.conditionalExpression(
-                    T.memberExpression(
-                        T.callExpression(T.identifier(this.varName), []),
-                        T.identifier(tempName)
+                    T.logicalExpression(
```

Comment:
This logic is a bit of a mouthful, would it be worth pulling into a utility method?

Reply:
Done